src/parse/index.ts

```diff
@@ -24,6 +24,10 @@ import {
   CssSupportsAST,
```

Comment:
suggestion: this a big file, maybe break it up into smaller modules? 

- parse/parser.ts
- parse/tokenizer.ts
- parse/ast-builder.ts

Reply:
suggestion: maybe add some css input validations(i.e. check string type, size, etc.)

Reply:
Yes, agree 👍 I'm not sure how the best way to break it down, but it will be better :-) I have already started to add some functions to another file for that purpose.

One thing to notice, though, performance is very important, so adding patterns like module class and so on will have a negative impact. So we should slowly find a way to split and refactor the file without introducing performance degradation, which makes it more challenging!